CS483/CS483/Kartaclysm/Components/ComponentKartController.cpp

```diff
@@ -125,30 +118,40 @@ namespace Kartaclysm
 		{
 			m_fSpeed -= m_fSpeed * m_fAccelerationFrictionStat * p_fDelta;
 		}
+	}
 
-		// Turning
-
+	void ComponentKartController::UpdateTurn(float p_fTurnInput, float p_fDelta)
+	{
+		// Determine slide direction, if sliding
 		if (m_bSliding)
 		{
-			if (fTurn < 0 && m_iSlideDirection >= 0)
+			if (p_fTurnInput < 0 && m_iSlideDirection == 0)
 			{
-				m_iSlideDirection = 1;
+				m_iSlideDirection = -1;
 			}
-			else if (fTurn > 0 && m_iSlideDirection <= 0)
+			else if (p_fTurnInput > 0 && m_iSlideDirection == 0)
 			{
-				m_iSlideDirection = -1;
+				m_iSlideDirection = 1;
 			}
 		}
 
-		float fTurnTarget = m_fMaxTurnStat * fTurn;
+		// Variables!
+		float fTurnTarget = m_fMaxTurnStat * p_fTurnInput;
 		float fModifier = 1.0f;
 
+		// Adjust turn parameters if sliding
 		if (m_bSliding)
```

Comment:
You're checking `m_bSliding == true` twice.  Move variable initialization to top of `UpdateTurn()` and then have a single if statement.